src/coreclr/jit/emitarm64.cpp

```diff
@@ -10571,6 +10601,48 @@ void emitter::emitIns_Call(EmitCallType          callType,
     return ureg << 10;
 }
 
+/*****************************************************************************
+ *
+ *  Returns an encoding for the specified register used in the 'Pd' position
+ */
+
+/*static*/ emitter::code_t emitter::insEncodeReg_Pd(regNumber reg)
+{
+    // TODO-SVE: Fix once we add predicate registers
+    assert(emitter::isPredicateRegister(reg));
+    emitter::code_t ureg = (emitter::code_t)reg - (emitter::code_t)REG_V0;
+    assert((ureg >= 0) && (ureg <= 15));
+    return ureg;
+}
+
+/*****************************************************************************
+ *
+ *  Returns an encoding for the specified register used in the 'Pn' position
+ */
+
+/*static*/ emitter::code_t emitter::insEncodeReg_Pn(regNumber reg)
```

Comment:
Seems a shame we can't reuse insEncodeReg_Zn etc - the function is very similar.

Reply:
The only differentiator is the following portion: Could use macro to share most of that code, but don't think it is worth it (for now).

```c++
    assert(emitter::isPredicateRegister(reg));
    emitter::code_t ureg = (emitter::code_t)reg - (emitter::code_t)REG_V0;
```

Reply:
I think it's not worth sharing. The asserts will be different even if the code is similar. It'll probably either get inlined or comdat folded in a release build anyway.